src/librustdoc/clean/types.rs

```diff
@@ -832,7 +832,10 @@ impl AttributesExt for [ast::Attribute] {
                 self.iter()
                     .filter(|attr| attr.has_name(sym::cfg))
                     .filter_map(|attr| single(attr.meta_item_list()?))
-                    .filter_map(|attr| Cfg::parse(attr.meta_item()?).ok())
+                    .filter_map(|attr| match Cfg::parse_without(attr.meta_item()?, &[sym::test]) {
+                        Ok(Some(c)) => Some(c),
+                        _ => None,
+                    })
                     .filter(|cfg| !hidden_cfg.contains(cfg))
```

Comment:
Couldn't we just add `test` into the `hidden_cfg`?

Reply:
Ah wait, that would need the other change to apply the hiding to sub-expressions rather than the expression as a whole.

Reply:
Yep, hence why I didn't. ;)

Reply:
I wonder if it would be good to make that change first, so we don't need the extra code here specifically to handle `test`?

Reply:
You mean putting `test` in `hidden_cfg` and then simply pass `hidden_cfg` to `Cfg::parse_without` instead of `&[sym::test]`?

Reply:
Yep, something like that.

Reply:
Ok!

Reply:
@Nemo157: That won't be possible as is because `cfg_hide` is to hide explicitly a `cfg` (`any(not(feature = "miri-test-libstd"), test, doctest)` for example) whereas this PR is to remove any appearances of `test`. The goals are completely different.

Reply:
Not really, `test` is just a very simple cfg-expression that could be hidden as more complex expressions can. As mentioned on the linked issue we can have `cfg_hide` hide arbitrary sub-expressions of the overall `cfg`. So e.g. if `test` is hidden `any(not(test), foo)` would simplify to `foo`, and if both `test` and `feature = "__internal"` are hidden then `any(test, feature = "__internal")` would simplify to nothing at all.

Reply:
Hum ok... Let me give it another try.